backend/src/routes/api/notebooks/utils.ts

```diff
@@ -27,10 +29,19 @@ export const getNotebookStatus = async (
   const notebookName = notebook?.metadata.name;
   let newNotebook: Notebook;
   if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
-    const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
-      fastify.log.warn(`Failed getting route ${notebookName}: ${e.message}`);
-      return undefined;
-    });
+    const disableServiceMesh = getDashboardConfig().spec.dashboardConfig.disableServiceMesh;
+    let route: Route;
+    if (!disableServiceMesh) {
```

Comment:
Here I'm seeing that same problem, this is based on the current status of the `OdhDashboardConfig`, but what happens when you enable/disable this feature and you've already deployed notebooks and projects?

Reply:
Because based on this piece of code, just based on the dashboard config, it will either try to get one route or the other, but this might happen for every notebook, not just the ones deployed with the feature enabled, would the itsio controller create routes for every notebook, even the ones. without the labels?

Reply:
Copied it to the backend and added it here 93aa3ae